src/overlays/actors/ovl_Bg_F40_Switch/z_bg_f40_switch.c

```diff
@@ -34,37 +35,160 @@ const ActorInit Bg_F40_Switch_InitVars = {
+        for (actor = globalCtx->actorCtx.actorList[0].first; actor; actor = actor->next) {
+            if (actor->id == ACTOR_BG_F40_SWITCH && actor->room == this->actor.actor.room && actor->update != 0) {
+                switchFlagIndex = (actor->params & 0xFE00) >> 9;
+                if (switchFlagIndex >= 0 && switchFlagIndex < 0x80 && Flags_GetSwitch(globalCtx, switchFlagIndex) &&
+                    !(pressedSwitchFlags[(switchFlagIndex & ~0x1F) >> 5] & (1 << (switchFlagIndex & 0x1F)))) {
+                    Actor_UnsetSwitchFlag(globalCtx, switchFlagIndex);
+                }
+            }
+        }
+        sBgF40SwitchLastUpdateFrame = globalCtx->gameplayFrames;
+    }
+}
+
+static InitChainEntry sInitChain[] = {
     ICHAIN_F32(uncullZoneForward, 4000, ICHAIN_CONTINUE),
     ICHAIN_F32(uncullZoneScale, 200, ICHAIN_CONTINUE),
     ICHAIN_F32(uncullZoneDownward, 200, ICHAIN_CONTINUE),
     ICHAIN_VEC3F_DIV1000(scale, 123, ICHAIN_STOP),
 };
 
-#endif
-
-extern InitChainEntry D_80BC4E04[];
-
-extern UNK_TYPE D_06000118;
-extern UNK_TYPE D_06000438;
-
-#pragma GLOBAL_ASM("asm/non_matchings/overlays/ovl_Bg_F40_Switch/func_80BC47B0.s")
-
-#pragma GLOBAL_ASM("asm/non_matchings/overlays/ovl_Bg_F40_Switch/BgF40Switch_Init.s")
-
-#pragma GLOBAL_ASM("asm/non_matchings/overlays/ovl_Bg_F40_Switch/BgF40Switch_Destroy.s")
-
-#pragma GLOBAL_ASM("asm/non_matchings/overlays/ovl_Bg_F40_Switch/func_80BC4B20.s")
-
-#pragma GLOBAL_ASM("asm/non_matchings/overlays/ovl_Bg_F40_Switch/func_80BC4B94.s")
-
-#pragma GLOBAL_ASM("asm/non_matchings/overlays/ovl_Bg_F40_Switch/func_80BC4BB8.s")
-
-#pragma GLOBAL_ASM("asm/non_matchings/overlays/ovl_Bg_F40_Switch/func_80BC4C68.s")
-
-#pragma GLOBAL_ASM("asm/non_matchings/overlays/ovl_Bg_F40_Switch/func_80BC4D30.s")
-
-#pragma GLOBAL_ASM("asm/non_matchings/overlays/ovl_Bg_F40_Switch/BgF40Switch_Update.s")
-
-#pragma GLOBAL_ASM("asm/non_matchings/overlays/ovl_Bg_F40_Switch/BgF40Switch_Draw.s")
+void BgF40Switch_Init(Actor* thisx, GlobalContext* globalCtx) {
+    BgF40Switch* this = THIS;
+
+    Actor_ProcessInitChain(&this->actor.actor, sInitChain);
+    this->actor.actor.scale.y = 0.165f;
+    this->actionFunc = BgF40Switch_ActionIdleNotPressed;
+    this->actor.actor.world.pos.y = this->actor.actor.home.pos.y + 1.0f;
+    DynaPolyActor_Init(&this->actor, 1);
+    DynaPolyActor_LoadMesh(globalCtx, &this->actor, &D_06000118);
+    if (!sBgF40SwitchGlobalsInitialized) {
+        sBgF40SwitchLastUpdateFrame = globalCtx->gameplayFrames;
+        sBgF40SwitchGlobalsInitialized = true;
+    }
+}
+
+void BgF40Switch_Destroy(Actor* thisx, GlobalContext* globalCtx) {
+    BgF40Switch* this = THIS;
+    DynaPoly_DeleteBgActor(globalCtx, &globalCtx->colCtx.dyna, this->actor.bgId);
+}
+
+void BgF40Switch_ActionBecomeNotPressed(BgF40Switch* this, GlobalContext* globalCtx) {
+    this->actor.actor.scale.y += 0.0495f;
+    if (this->actor.actor.scale.y >= 0.165f) {
+        Audio_PlayActorSound2(&this->actor.actor, 0x29A6U);
```

Comment:
There are defines for these in sfx.h
```suggestion
        Audio_PlayActorSound2(&this->actor.actor, NA_SE_EV_IKANA_BLOCK_SWITCH);
```